beta/src/content/apis/react/Component.md

```diff
@@ -1,28 +1,1797 @@
 ---
-title: React.Component
+title: Component
 ---
 
-<Wip>
+<Pitfall>
 
-This section is incomplete, please see the old docs for [React.Component.](https://reactjs.org/docs/react-component.html)
-
-</Wip>
+We recommend to define components as functions instead of classes. [See how to migrate.](#alternatives)
 
+</Pitfall>
 
 <Intro>
 
-React lets you define components as classes:
+`Component` is the base class for the React components defined as [JavaScript classes.](https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Classes) Class components are still supported by React, but we don't recommend using them in new code.
 
 ```js
-class Welcome extends React.Component {
+class Greeting extends Component {
   render() {
-    return <h1>Hello, {this.props.name}</h1>;
+    return <h1>Hello, {this.props.name}!</h1>;
   }
 }
 ```
 
-The only method you must define in a `React.Component` subclass is called `render()`. All the other methods described on are optional. Please see the old [React.Component](https://reactjs.org/docs/react-component.html) docs for more info.
-
 </Intro>
 
 <InlineToc />
+
+---
+
+## Usage {/*usage*/}
+
+### Defining a class component {/*defining-a-class-component*/}
+
+To define a React component as a class, extend the built-in `Component` class and define a [`render` method:](#render)
+
+```js
+import { Component } from 'react';
+
+class Greeting extends Component {
+  render() {
+    return <h1>Hello, {this.props.name}!</h1>;
+  }
+}
+```
+
+React will call your [`render`](#render) method whenever it needs to figure out what to display on the screen. Usually, you will return some [JSX](/learn/writing-markup-with-jsx) from it. Your `render` method should be a [pure function:](https://en.wikipedia.org/wiki/Pure_function) it should only calculate the JSX.
+
+Similarly to [function components,](/learn/your-first-component#defining-a-component) a class component can [receive information by props](/learn/your-first-component#defining-a-component) from its parent component. However, the syntax for reading props is different. For example, if the parent component renders `<Greeting name="Taylor" />`, then you can read the `name` prop from [`this.props`](#props), like `this.props.name`:
+
+<Sandpack>
+
+```js
+import { Component } from 'react';
+
+class Greeting extends Component {
+  render() {
+    return <h1>Hello, {this.props.name}!</h1>;
+  }
+}
+
+export default function App() {
+  return (
+    <>
+      <Greeting name="Sara" />
+      <Greeting name="Cahal" />
+      <Greeting name="Edite" />
+    </>
+  );
+}
+```
+
+</Sandpack>
+
+Note that Hooks (functions starting with `use`, like [`useState`](/apis/react/useState)) are not supported inside class components.
+
+<Pitfall>
+
+We recommend to define components as functions instead of classes. [See how to migrate.](#migrating-a-simple-component-from-a-class-to-a-function)
```

Comment:
<img width="361" alt="Screenshot 2022-11-15 at 11 31 35 PM" src="https://user-images.githubusercontent.com/32865581/201992750-77953dad-d6d0-4c0e-9f9b-4d404d4232e6.png">
This text is almost 5 times in this page. Should we impose it to this extent? I feel we could do it once on top alone?

Reply:
no, I want it to be in-your-face

Reply:
🖖🏻